### PR TITLE
Refactor three xfail grdcontour tests

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Refractor three grdcontour tests using the `@check_figures_equal` decorator so that they no longer xfail.

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

### CODE REVIEW

File: pygmt/tests/test_grdcontour.py [hunk 1]
```diff
@@ -21,49 +23,53 @@ def fixture_grid():
...
-    return fig
+    fig_test.grdcontour(grid_, interval="1000", projection="M6i")
+
+    with GMTTempFile(suffix=".nc") as tmpgrid:
+        grdcut(
+            grid="@earth_relief_01d_g",
+            region=[-180, 180, -30, 30],
+            outgrid=tmpgrid.name,
+        )
+        fig_ref.grdcontour(tmpgrid.name, interval="1000", projection="M6i")
```
Comment (weiji14) [inline pygmt/tests/test_grdcontour.py:71]:
```suggestion
    kwargs = dict(interval="1000", projection="M6i")
    fig_ref.grdcontour(
        grid="@earth_relief_01d_g", region=[-180, 180, -30, 30], **kwargs
    )
    fig_test.grdcontour(grid=grid_, **kwargs)
```
`grdcut` and `GMTTempFile` won't be needed here :)

File: pygmt/tests/test_grdcontour.py [hunk 2]
```diff
@@ -6,9 +6,11 @@
 import numpy as np
 import pytest
 
-from .. import Figure
+from .. import Figure, grdcut
 from ..exceptions import GMTInvalidInput
 from ..datasets import load_earth_relief
+from ..helpers import GMTTempFile
```
Comment (weiji14) [inline pygmt/tests/test_grdcontour.py:12]:
```suggestion
from .. import Figure
from ..exceptions import GMTInvalidInput
from ..datasets import load_earth_relief
```

**Review by weiji14 — COMMENTED** [review]
Thanks for keeping on top of this (sorry that I haven't been keeping up much, got a bit busy with PhD work again)! Just one suggestion to simplify the refactor.

Also, there's another xfail for `test_grdcontour_interval_file_full_opts` that wasn't removed, is there a reason to leave it?

**Comment by seisman** [comment]
> there's another xfail for `test_grdcontour_interval_file_full_opts` that wasn't removed, is there a reason to leave it?

This test uses the grid file `@earth_relief_10m`, so we can't rewrite the test using `@check_figure_equal()` decorator. To remove the xfail label, we need to update the baseline image, but we're trying to avoid it before we have a better solution.

**Comment by weiji14** [comment]
> This test uses the grid file `@earth_relief_10m`, so we can't rewrite the test using `@check_figure_equal()` decorator. To remove the xfail label, we need to update the baseline image, but we're trying to avoid it before we have a better solution.

Yeah, doesn't look like a trivial task. Happy to leave it for now then.

**Review by weiji14 — APPROVED** [review]

**Pull request merged**